### DISCUSSION TITLE
know which checkbox called the function (on_change)

### DISCUSSION THREAD

**Discussion by pasteluengas** [flet-dev/flet#964] [Q&A] [answered]
Hi, suppose i have these checkboxes

`ft.DataColumn(ft.Checkbox(value=False, on_change=lambda e: checkbox_changed(kick_pattern, 0))),
ft.DataColumn(ft.Checkbox(value=False, on_change=lambda e: checkbox_changed(kick_pattern, 0))),
ft.DataColumn(ft.Checkbox(value=False, on_change=lambda e: checkbox_changed(kick_pattern, 0))),
ft.DataColumn(ft.Checkbox(value=False, on_change=lambda e: checkbox_changed(kick_pattern, 0))),
ft.DataColumn(ft.Checkbox(value=False, on_change=lambda e: checkbox_changed(kick_pattern, 0))),`

checkbox_changed function is:

`def checkbox_changed(pattern, val):` 
`       # Insert here something here that depends to the value of the checkbox that called it`

I need to know which checkbox called the function, to get the `.value` from it... Any idea?

Thanks

**Comment by iron3oxide** [comment, marked as answer]
The event `e` your lambdas catch will hold a reference to the control that emitted it. You could therefore do something like 
```python
callback_fn = lambda e: checkbox_changed(kick_pattern, e.control.value)
...
ft.DataColumn(ft.Checkbox(value=False, on_change=callback_fn)),
...
```
